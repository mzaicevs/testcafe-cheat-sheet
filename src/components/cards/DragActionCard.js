import React from "react";
import { actions_drag } from "../../snippets";
import CodeCard from "../../base_components/CodeCard";

const DragActionCard = () => (
  <CodeCard
    cardTitle={actions_drag.title}
    fullDocLink={actions_drag.fullDocLink}
    codeSnippet={{
      code: actions_drag.code,
      language: actions_drag.language,
      options: actions_drag.options,
      example: actions_drag.example
    }}
    description={actions_drag.description}
  />
);

export default DragActionCard;
