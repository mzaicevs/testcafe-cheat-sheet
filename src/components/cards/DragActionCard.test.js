import React from "react";
import { shallow } from "enzyme";
import DragActionCard from "./DragActionCard";

jest.mock("react-syntax-highlighter/dist/esm/styles/hljs", () => () => {});

describe("ClickActionCard", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<DragActionCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
