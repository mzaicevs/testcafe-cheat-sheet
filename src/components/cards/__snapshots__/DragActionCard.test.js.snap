// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickActionCard matches snapshot 1`] = `
<CodeCard
  cardTitle={
    <CardLink
      href="https://devexpress.github.io/testcafe/documentation/test-api/actions/drag-element.html"
      text="Drag Action"
    />
  }
  codeSnippet={
    Object {
      "code": Array [
        "t.drag( selector, dragOffsetX, dragOffsetY [, options] )",
        "t.dragToElement( selector, destinationSelector [, options] )",
      ],
      "example": "import { Selector } from 'testcafe';

  const slider = Selector('#developer-rating');
  
  fixture \`My fixture\`
      .page \`http://www.example.com/\`;
  
  test('Drag slider', async t => {
      await t
          .click('#i-tried-testcafe');
          .expect(slider.value).eql(1)
          .drag('.ui-slider-handle', 360, 0, { offsetX: 10, offsetY: 10 })
          .expect(slider.value).eql(7);
          .dragToElement('.ui-slider-handle', '.developer-rating-best')
          .expect(slider.value).eql(10);
  });",
      "language": "javascript",
      "options": "{
      modifiers: {
          ctrl: Boolean,
          alt: Boolean,
          shift: Boolean,
          meta: Boolean
      },
  
      offsetX: Number,
      offsetY: Number,
      destinationOffsetX: Number,
      destinationOffsetY: Number,
      speed: Number
  }",
    }
  }
  description="Drag a webpage element in a test."
/>
`;
